src/ucp/tag/eager_rcv.c

```diff
@@ -259,6 +259,7 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_eager_offload_sync_ack_handler,
     ucs_queue_for_each_safe(sreq, iter, queue, send.tag_offload.queue) {
         if ((sreq->send.tag_offload.ssend_tag == rep_hdr->sender_tag) &&
             (ucp_ep_local_id(sreq->send.ep) == rep_hdr->ep_id)) {
+            ucp_request_id_release(sreq);
```

Comment:
aren't error handling and tag offload mutually exclusive?

Reply:
yes, they are mutually exclusive, but better to release ID even if it was set to direct pointer
also, to be compliant with `ucp_request_complete()`/`ucp_request_put()` which checks request ID

Reply:
do we have paired ucp_request_id_alloc? Looks like we do not

Reply:
we have it here - https://github.com/openucx/ucx/blob/6e258e7550df0feb9ee40c2dee388cd6e233d4f9/src/ucp/tag/tag_send.c#L92

Reply:
ok